src/asgen/backends/alpinelinux/apkpkgindex.d

```diff
@@ -0,0 +1,202 @@
+            }
+
+            ad.open(packageArchivePath);
+            auto pkgInfoData = cast(string) ad.readData(".PKGINFO");
+
+            try
+            {
+                validate(pkgInfoData);
+            }
+            catch (UTFException e)
+            {
+                logError("PKGINFO file in archive %s contained invalid UTF-8, skipping!",
+                        packageArchivePath);
+            }
+
+            pkg.filename = packageArchivePath;
+            auto lines = pkgInfoData.splitLines();
+            for (auto i = 0; i < lines.length; i++)
+            {
+                /* Skip comments */
+                if (lines[i].startsWith("#"))
+                {
+                    continue;
+                }
+
+                auto line = lines[i].strip().split("#")[0];
+
+                /*
+                    When the next line doesn't contain a "=" it's a continuation
+                    of the current line
+                */
+                int j = i + 1;
+                while (lines.length > j && !lines[j].canFind("=") && !lines[j].startsWith("#"))
+                {
+                    line ~= " " ~ lines[j].strip().split("#")[0];
+                    /* Skip the next line since it was a continuation of the previous one */
+                    i++;
+                    j++;
+                }
+
+                auto infoEntry = line.split("=");
+                immutable key = infoEntry[0].strip();
+                immutable auto value = infoEntry[1].strip();
+                switch (key)
+                {
+                case "pkgname":
+                    pkg.name = value;
+                    break;
+                case "pkgver":
+                    pkg.ver = value;
+                    break;
+                case "arch":
+                    pkg.arch = value;
+                    break;
+                case "maintainer":
+                    pkg.maintainer = value;
+                    break;
+                case "pkgdesc":
+                    setPkgDescription(pkg, value);
+                    break;
+                default:
+                    /* We don't care about other entries */
+                    break;
+                }
+            }
+            pkg.contents = ad.readContents();
+        }
+
+        // perform a sanity check, so we will never emit invalid packages
+        auto pkgs = appender!(Package[]);
+        if (pkgsMap.length > 20)
+            pkgs.reserve((pkgsMap.length.to!long - 10).to!size_t);
+        foreach (ref pkg; pkgsMap.byValue())
+        {
+            if (pkg.isValid)
+                pkgs ~= pkg;
+            else
+                logError(
+                        "Found an invalid package (name, architecture or version is missing). This is a bug.");
+        }
+
+        return pkgs.data;
+    }
+
+    override Package[] packagesFor(string suite, string section, string arch,
+            bool withLongDescs = true)
```

Comment:
style: Please align parameters after a line break with the other parameters:
``` 
override Package[] packagesFor(string suite, string section, string arch,
                               bool withLongDescs = true)
{
```
In this case you probably don't even need to break the line though (with D's ultra-verbose function decorators, having space here is a good thing)